src/services/auth/oidc/loginAttempts/types.ts

```diff
@@ -0,0 +1,72 @@
+import { OptionsWithRql, PagedResultWithPager } from '../../../types';
+import { FindAllIterator } from '../../../helpers';
+
+export interface LoginAttemptsService {
+  /**
+   * ## Retrieve a paged list of login attempts
+   *
+   * **Global Permissions:**
+   * - `VIEW_OIDC_LOGIN_ATTEMPTS` - Allows a user to view login attempts
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns A paged list of login attempts {@link PagedResultWithPager PagedResultWithPager<LoginAttempt>}
+   */
+  find(options?: OptionsWithRql): Promise<PagedResultWithPager<LoginAttempt>>;
+
+  /**
+   * ## Retrieve a list of all login attempts
+   *
+   * **Global Permissions:**
+   * - `VIEW_OIDC_LOGIN_ATTEMPTS` - Allows a user to view login attempts
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns An array of login attempts {@link LoginAttempt LoginAttempt[]}
+   * @throws {@link Error} Do not pass in limit operator with findAll
+   */
+  findAll(options?: OptionsWithRql): Promise<LoginAttempt[]>;
+
+  /**
+   * ## Retrieve a paged list of login attempts
+   *
+   * **Global Permissions:**
+   * - `VIEW_OIDC_LOGIN_ATTEMPTS` - Allows a user to view login attempts
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns An iterator for the queried login attempts {@link FindAllIterator FindAllIterator<LoginAttempt>}
+   */
+  findAllIterator(options?: OptionsWithRql): FindAllIterator<LoginAttempt>;
+
+  /**
+   * ## Retrieve a list of login attempts
+   *
+   * **Global Permissions:**
+   * - `VIEW_OIDC_LOGIN_ATTEMPTS` - Allows a user to view login attempts
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns The first element of the queried login attempts {@link LoginAttempt}
+   */
+  findFirst(options?: OptionsWithRql): Promise<LoginAttempt>;
+}
+
+export enum LoginAttemptStatus {
+  SUCCESS = 'success',
+  FAILED = 'failed',
+}
+
+export interface LoginAttempt {
+  /** The Extra Horizon document id */
+  id: string;
+  /** The (success / failed) status of a login attempt. */
+  status: LoginAttemptStatus;
+  /** The id of the provider */
+  providerId: string;
+  /** The unique name of the provider */
+  providerName: string;
+  /** The subject id (sub) of the user for the given provider */
+  providerSubjectId: string;
+  /** The Extra Horizon user id */
+  userId: string;
+  /** An Extra Horizon formatted error for failed requests */
+  error: {
+    name: string;
+    message: string;
+    code: number;
+  };
+  creationTimestamp: Date;
```

Comment:
Do we want to add documentation for the normal returned fields `id`, `creationTimestamp` etc... ? I'm not sure that they need explanation let alone multiple copy pastes.

Reply:
`id` would indeed be one we could skip. In most cases it could not hurt to document the `creationTimestamp` though. In this case for instance, it might be good to make it extra clear and specify that this is indeed "the time the login attempt happened".